flow/util/genRuleFile.py

```diff
@@ -189,10 +181,7 @@ def gen_rule_file(
     change_str = ""
     for field, option in rules_dict.items():
         if field not in metrics.keys():
-            print(
-                f"[ERROR] Metric {field} not found in "
-                f"metrics file: {metrics_file} or golden metrics."
-            )
+            print(f"[ERROR] Metric {field} not found")
```

Comment:
Printing the name of the file is helpful because we have utilities in the CI that run this in parallel.

Reply:
Can you explain a bit more?

This would be inconsistent with all other error messages in OpenROAD/ORFS.

Error and progress messages do not contain context in which they run, because it doesn't scale.

Rather the CI system has to present log files in a way that only a single job and context is shown and that output from multiple jobs are not intermingled.

If we are to address this, then I would:

1. switch to exceptions (context and stack trace)
2. add a finally statement that prints out some extra information, like file name in the event of an exception

However, I'm unsure this is a good idea. A good CI system would give adequate context that it would be obvious what is being processed and the extra logging cruft adds a maintenance burden and clutters the code.

Reply:
We don't have the design name, we only have input and output .json rule files.

Also, the command line is logged before the output, which gives context.